### ISSUE TITLE
Share full article to use TTS

### ISSUE THREAD

**Issue by X-dark** [FreshRSS/FreshRSS#4760]
Hi,
Is there a way to share the full article content and not just the url? It would be useful when the content is behind a paywall, to share easily to family members a specific article.

**Comment by aledeg** [comment]
Probably. The share system must be improved though. Help welcomed.

**Comment by math-GH** [comment]
Could you please give a concrete use case?

If the content is behind a paywall, how do you fetch the content?
How do you want to send the content? How to handle images within the content? How to handle articles with 10.000 words?

**Comment by X-dark** [comment]
Either the website is providing a feed for subscriber (eg NextInpact) or a dedicated script is fetching the full article with my credential from the abridged rss feed (eg Telerama). The issue is not really there.

In any case, the full article is there in FreshRSS. One of the two following would be satisfying for me:

* share to email is sending the full article (plain text or even better in HTML)
* an external link can be generated to access the article the way it was fetched by FreshRSS, with anonymous access

**Comment by aledeg** [comment]
If you want to share that, FreshRSS provides RSS feeds for categories, feeds, searches (to be confirmed), and labels.
You could add a label to those articles and share the associated feed.

**Comment by math-GH** [comment]
I have doubt that it is easy possible to "share" text via links with a lot of characters. I guess that there is often a 255 chars limit.

**Comment by X-dark** [comment]
> I have doubt that it is easy possible to "share" text via links with a lot of characters. I guess that there is often a 255 chars limit.

Yes, hence my second idea. For the first one, it was more with Freshrss feeding the content to Sendmail.

**Comment by alfureu** [comment]
> Could you please give a concrete use case?
> 
> If the content is behind a paywall, how do you fetch the content? How do you want to send the content? How to handle images within the content? How to handle articles with 10.000 words?

Another use case to the already provided is using read-aloud services, for example, on Android devices. I often swipe through articles while waiting for train or on a street. If I find something interesting, I either mark it to read later or if time allows, while need to do sth else, put it on headphones to read it aloud. 

AFAIK, there is no character limit in sharing on Android

**Comment by math-GH** [comment]
> AFAIK, there is no character limit in sharing on Android

In our case: The max length is a question of browser and server configs.

All sharing services in FreshRSS use the GET parameters for sharing (the GET parameters are part of the URL). That is right now IMHO the limiting factor.

Some example for max URL length (source: https://www.sistrix.com/ask-sistrix/technical-seo/site-structure/url-length-how-long-can-a-url-be)

-     Microsoft Internet Explorer: 2,083 characters
-     Microsoft Edge: 2,083 characters
-     Google Chrome: 32,779 characters
-     Mozilla Firefox: more than 64,000 characters
-     Apple Safari: more than 64,000 characters
-     Google Android: 8,192 characters
-     Apache: 8,177 characters
-     NGINX: 4,096 characters
-     Microsoft IIS: 16,384 characters
-     Fastly (CDN): 8,192 characters
-     Amazon Cloudfront (CDN): 8,192 characters
-     Cloudflare (CDN): 32,768 characters

**Comment by alfureu** [comment]
Interesting, is it possible to maybe use the API access for this? For example, I am currently circumventing the limitation on Android by using an RSS client that forwards the text to the TTS. It works even with articles that are paywalled but I have paid access to them. Sth similar would be nice to have on desktop as well. 

EDIT: just to clarify, I am not talking about sharing the URL, but the text of the article as shown in the FreshRSS UI

**Comment by math-GH** [comment]
>  I am currently circumventing the limitation on Android by using an RSS client that forwards the text to the TTS.

Could you please describe which TTS (text to speech?) app you use? It would help to understand it better, when I could try it by myself.

I am also interested in your use case of TTS. Do you use it often?

**Comment by alfureu** [comment]
I connect to FreshRSS with Readably, from there I share the text of the article to the app called @Voice Aloud Reader
